test/extension/config/default/manager.yaml

```diff
@@ -2,24 +2,24 @@
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: test-extension
+  name: manager
```

Comment:
That's called "controller-manager" in other providers

Reply:
this is similar to https://github.com/kubernetes-sigs/cluster-api/pull/7288#discussion_r981587836 (clusterctl contract is not prescriptive on this name)